save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt

```diff
@@ -28,4 +31,32 @@ internal class OrganizationController(private val organizationService: Organizat
      */
     @GetMapping("/avatar")
     fun avatar(@RequestParam owner: String): ImageInfo? = organizationService.findByName(owner).avatar.let { ImageInfo(it) }
+
+    /**
+     * @param newOrganization newOrganization
+     * @param authentication an [Authentication] representing an authenticated request
+     * @return response
+     */
+    @PostMapping("/save")
+    fun saveOrganization(@RequestBody newOrganization: Organization, authentication: Authentication): ResponseEntity<String> {
+        val ownerId = (authentication.details as AuthenticationDetails).id
+        val (organizationId, organizationStatus) = organizationService.saveOrganization(
+            newOrganization.apply {
+                this.ownerId = ownerId
+                this.dateCreated = LocalDateTime.now()
+            }
+        )
+
+        if (organizationStatus == OrganizationSaveStatus.EXIST) {
+            logger.warn("Organization with id = $organizationId already exists")
```

Comment:
Will any data get overwritten in this case?

Reply:
No, it will not because if there is an existing organization that conflicts the new one, an existing one will be returned from `saveOrganization`.

Reply:
Right, we are checking if it exists inside the method. Maybe, then rename it to something like `getOrSaveOrganization`?

Reply:
Maybe but we need to rename same method for `ProjectController`. I guess it would be nice to keep same format for same things in different classes.